dataTypes/example.go

```diff
@@ -2,40 +2,31 @@ package dataTypes
 
 import (
 	"encoding/json"
-	"fmt"
 )
-
+type Person struct {
+	FirstName string `json:"firstName"`
+	LastName  string `json:"lastName"`
+}
 func Example() {
-	email := Field{}
-	email.Type = "string"
-	email.Field = "VendorMergedEmail"
-	email.DisplayName = "Email"
-
-	dob := Field{}
-	dob.Type = "string"
-	dob.Field = "VendorDateOfBirth"
-	dob.DisplayName = "Date of Birth"
-
-
-	candidate := Candidate{}
-	candidate.Email = email
-	candidate.Dob = dob
-
-	fieldMap := FieldMap{}
-	fieldMap.Candidate = candidate
-
-	dataJSON := `{
-	   "VendorDateOfBirth": "11/12/13",
-	   "VendorMergedEmail": "dixon@awesome.co.uk"
-	}`
-
-	// Declared an empty map interface
-	var result map[string]string
-
-	// Unmarshal or Decode the JSON to the interface.
-	json.Unmarshal([]byte(dataJSON), &result)
-
-	// Run App Migrations
-	ret := ToElectiveStruct(fieldMap, result)
-	fmt.Print(ret)
+	fieldMapJSON := `{"candidate":{"Dob":{"type":"string","field":"VendorDateOfBirth","DisplayName":"Date of Birth"},"Email":{"type":"string","field":"VendorMergedEmail","DisplayName":"Email"}}}`
+
+	dataJSON := `{"VendorDateOfBirth":"11/12/13","VendorMergedEmail":"dixon@awesome.co.uk"}`
+
+	rawData := json.RawMessage(dataJSON)
+	data, err := rawData.MarshalJSON()
+	if err != nil {
+		panic(err)
+	}
+
+	rawField := json.RawMessage(fieldMapJSON)
+	fieldMap, err := rawField.MarshalJSON()
+	if err != nil {
+		panic(err)
```

Comment:
😱 panic and run for cover ... program terminated ;)

Reply:
yeah this was work in progress, panic removed in later MR